viewing_party/party.py

```diff
@@ -1,23 +1,162 @@
 # ------------- WAVE 1 --------------------
+from platform import libc_ver
+
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if title != None:
+        new_movie["title"] = title
+        if genre != None:
+            new_movie["genre"] = genre
+            if rating != None:
+                new_movie["rating"] = rating
+                return new_movie
+            else:
+                return None
+        else:
+            return None
+    else:
+        return None
+            
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for i in range(len(user_data["watchlist"])):
+        if title == user_data["watchlist"][i]["title"]: 
+            user_data["watched"].append(user_data["watchlist"][i])
+            user_data["watchlist"].pop(i)
+            
+            return user_data
+    return user_data
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    if len(user_data["watched"]) >= 1:
+        sum = 0
+        for i in range(len(user_data["watched"])):
+            sum += user_data["watched"][i]["rating"]
+        return sum / len(user_data["watched"])   
+    else:
+        return 0.0
 
-
+def get_most_watched_genre(user_data):
+    if len(user_data["watched"]) >= 1:
+        genre_list = []
+        for i in range(len(user_data["watched"])):
+            genre_list.append(user_data["watched"][i]["genre"])
+        
+        counter = 0
+        most_frequent_genre = genre_list[0]
+        for i in genre_list:
+            curr_frequency = genre_list.count(i)
+            if(curr_frequency> counter):
+                counter = curr_frequency
+                most_frequent_genre = i
+ 
+        return most_frequent_genre

```

Comment:
This implementation works, but does a bit more work than necessary. Since `for i in genre_list` will go through all elements in `genre_list`, that means it will check multiple times if there are repeated genres.

While y'all hadn't been introduced to Big O notation when you wrote this, there is a performance downside that can be highlighted as such. The `for` loop is going to go through the n genres in `genre_list`, where n is the number of movies in `user_data["watched"]`, which is O(n). For each of these n genres, it will call `genre_list.count(i)`, which is itself O(n). Combined this makes it O(n^2).

Rather than making a list of the genres, you can use a dictionary to keep track of the frequencies of the genres and find the answer in O(n) time. You're not required to do so, but I do recommend trying to implement that answer as it is a very common pattern.